naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionField.scala

```diff
@@ -48,8 +49,13 @@ object NaptimeUnionField {
         case definitions: java.util.Map[String @unchecked, String @unchecked] => definitions.asScala
       }.getOrElse(Map[String, String]())
 
-      val unionMemberKey = typedDefinitions
-        .getOrElse(subType.getUnionMemberKey, subType.getUnionMemberKey)
+      val unionMemberKey = (subType, typedDefinitions) match {
+        case (x,y) if y.contains(x.getUnionMemberKey) =>
+          typedDefinitions(x.getUnionMemberKey)
+        case (x: RecordDataSchema, y) if y.contains(x.asInstanceOf[RecordDataSchema].getName) =>
+          y(x.asInstanceOf[RecordDataSchema].getName)
```

Comment:
if you're matching against `RecordDataSchema`, you probably don't need to use `.asInstanceOf`